app/src/components/dataset/EditDatasetForm.tsx

```diff
@@ -0,0 +1,188 @@
+          description
+          metadata
+        }
+      }
+    }
+  `);
+
+  const onSubmit = (params: PatchDatasetParams) => {
+    commit({
+      variables: {
+        datasetId,
+        ...params,
+        metadata: JSON.parse(params.metadata),
+      },
+      onCompleted: () => {
+        onDatasetEdited();
+      },
+      onError: (error) => {
+        onDatasetEditError(error);
+      },
+    });
+  };
+
+  return (
+    <Form>
+      <View padding="size-200">
+        <Controller
+          name="name"
+          control={control}
+          rules={{
+            required: "Dataset name is required",
+          }}
+          render={({
+            field: { onChange, onBlur, value },
+            fieldState: { invalid, error },
+          }) => (
+            <TextField
+              label="Dataset Name"
+              description={`The name of the dataset`}
+              errorMessage={error?.message}
+              validationState={invalid ? "invalid" : "valid"}
+              onChange={onChange}
+              onBlur={onBlur}
+              value={value.toString()}
+            />
+          )}
+        />
+        <Controller
+          name="description"
+          control={control}
+          render={({
+            field: { onChange, onBlur, value },
+            fieldState: { invalid, error },
+          }) => (
+            <TextArea
+              label="description"
+              isRequired={false}
+              height={100}
+              errorMessage={error?.message}
+              validationState={invalid ? "invalid" : "valid"}
+              onChange={onChange}
+              onBlur={onBlur}
+              value={value?.toString()}
+            />
+          )}
+        />
+        <Controller
+          name="metadata"
+          control={control}
+          rules={{
+            validate: (value) => {
+              if (!isJSONObjectString(value)) {
+                return "metadata must be a valid JSON object";
+              }
+              return true;
+            },
+          }}
+          render={({
+            field: { onChange, onBlur, value },
+            fieldState: { invalid, error },
+          }) => (
+            <div css={metadataFieldWrapperCSS}>
+              <Field
+                label={"metadata"}
+                validationState={invalid ? "invalid" : "valid"}
+                errorMessage={error?.message}
+              >
+                <JSONEditor
+                  value={value}
+                  onChange={onChange}
+                  onBlur={onBlur}
+                  width="100%"
+                />
+              </Field>
+            </div>
+          )}
+        />
+      </View>
+      <View
+        paddingEnd="size-200"
+        paddingTop="size-100"
+        paddingBottom="size-100"
+        borderTopColor="light"
+        borderTopWidth="thin"
+      >
+        <Flex direction="row" justifyContent="end">
+          <Button
+            disabled={!isDirty}
```

Comment:
isDisabled prop doesn't actually disable button - only allow for clicking save when fields have changed, invalid will be caught by useForm rules

Reply:
Yeah I noticed that too the other day. We should fix in the component lib

Reply:
yeah both arize and phoenix are all over the place isDisabled / disable